### PR TITLE
Telemetry test

### PR DESCRIPTION
**Pull request by RichardPen**
Added telemetry functional test.

### CODE REVIEW

**Comment by aaithal** [comment]
:shipit:

**Comment by samuelkarp** [comment]
I ran your new test with `make run-functional-tests` and it seems to be timing out:

```
=== RUN TestTelemetry
panic: test timed out after 20m0s
```

It looks (to me) like the timeout is [here](https://github.com/RichardPen/amazon-ecs-agent/blob/63c6547a689937235db3d5a21c09b23ebc035b64/agent/functional_tests/tests/functionaltests_test.go#L635), so we may need to bump the test timeout above 20 minutes.

File: agent/functional_tests/testdata/taskdefinitions/telemetry/task-definition.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "family": "ecs-metrics-test",
+    "containerDefinitions": [{
+        "memory": 100,
+        "name": "telemetry",
+        "cpu": 100,
+        "image": "ubuntu:latest",
```
Comment (samuelkarp) [inline agent/functional_tests/testdata/taskdefinitions/telemetry/task-definition.json]: @aaithal thanks for pointing this out to me!

@RichardPen, I think the pull time for `ubuntu:latest` on a host with no cached images is contributing to the test timeout.  You'll want to do the same thing that we've done for our other test images.

Here's an example [task definition](https://github.com/aws/amazon-ecs-agent/blob/v1.8.1/agent/functional_tests/testdata/taskdefinitions/datavolume/task-definition.json#L18) with the image set to a registry running on `localhost` (started with our [`setup-test-registry`](https://github.com/aws/amazon-ecs-agent/blob/v1.8.1/scripts/setup-test-registry) script).  You'll also need to modify the script to add the `ubuntu` image [like this](https://github.com/aws/amazon-ecs-agent/blob/v1.8.1/scripts/setup-test-registry#L71-L77).

**Comment by samuelkarp** [comment]
It looks like it's still failing even with the caching.  Let's dig into this together tomorrow (and maybe just lengthen the time it takes to run).

**Comment by samuelkarp** [comment]
:+1:

**Comment by samuelkarp** [comment]
Closed by https://github.com/aws/amazon-ecs-agent/commit/5b23f91239903e62c7b1461c57bba76b737e9097 (commit got squashed too much, but this the same content).

**Pull request closed without merging**